core/src/main/scala/org/apache/spark/api/python/PythonRunner.scala

```diff
@@ -401,33 +415,35 @@ private[spark] abstract class BasePythonRunner[IN, OUT](
             }
           }.start()
         }
-        val secret = if (isBarrier) {
-          authHelper.secret
-        } else {
-          ""
-        }
         if (isBarrier) {
           // Close ServerSocket on task completion.
-          serverSocket.foreach { server =>
-            context.addTaskCompletionListener[Unit](_ => server.close())
+          serverSocketChannel.foreach { server =>
+            context.addTaskCompletionListener[Unit] { _ =>
+              server.close()
+              if (isUnixDomainSock) sockPath.deleteOnExit()
```

Comment:
Do we still need to call `sockPath.deleteOnExit()`? It's already called at [line 367](https://github.com/apache/spark/pull/50466/files#diff-ed4fb5ce30273e8eefcc7d4b0152ea7a60fb4f8f709d4da8ea1ab56aeda26001R367)?
Or it should be `sockPath.delete()` instead? May want to consider to use `Files.deleteIfExists`.

Reply:
Yeah, it should be `delete()`. `File.delete` won't fail anyway.